pkg/plugins/common/kustomize/v2-alpha/scaffolds/internal/templates/config/samples/kustomization.go

```diff
@@ -17,54 +17,60 @@ limitations under the License.
 package samples
 
 import (
-	"bytes"
 	"fmt"
 	"path/filepath"
-	"text/template"
-
 	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
 )
 
-var _ machinery.Template = &Kustomization{}
+var (
+	_ machinery.Template = &Kustomization{}
+	_ machinery.Inserter = &Kustomization{}
+)
 
-// Kustomization scaffolds a file that defines the kustomization scheme for the prometheus folder
+// Kustomization scaffolds a kustomization.yaml for the manifests overlay folder.
 type Kustomization struct {
 	machinery.TemplateMixin
-
-	CRDManifests []string
+	machinery.ResourceMixin
 }
 
-// SetTemplateDefaults implements file.Template
+// SetTemplateDefaults implements machinery.Template
 func (f *Kustomization) SetTemplateDefaults() error {
 	if f.Path == "" {
 		f.Path = filepath.Join("config", "samples", "kustomization.yaml")
 	}
+	f.Path = f.Resource.Replacer().Replace(f.Path)
```

Comment:
I'm wondering why we'd like to updated `f.Path` here? Will it be values other than `config/samples/kustomization.yaml` by calling `replacer`?

Reply:
Good catcher, we do not need this here.